src/main/java/com/targus/ui/widgets/Sensor.java

```diff
@@ -8,19 +8,56 @@
 public class Sensor extends Group {
     private static double COMMUNICATION_RADIUS;
     private static double SENSING_RADIUS;
+    private static boolean STATIC_VARIABLES_INITIALIZED = false;
+    private Circle sensor;
+
+    public static void initializeRadii(double communicationRadius, double sensingRadius) {
+        setCommunicationRadius(communicationRadius);
+        setSensingRadius(sensingRadius);
+        STATIC_VARIABLES_INITIALIZED = true;
+    }
+
+    private static boolean staticVariablesInitialized() {
+        return COMMUNICATION_RADIUS != 0.0
+                && SENSING_RADIUS != 0.0;
+    }
 
     public Sensor(double centerX, double centerY) {
-        Circle circle = new Circle(centerX, centerY, 4);
-        circle.setFill(Color.GREEN);
+        if (!staticVariablesInitialized()) {
+            throw new IllegalStateException("Radii are not initialized.");
+        }
+
+        sensor = new Circle(centerX, centerY, 4);
```

Comment:
We can parameterize the magic number. In `Constants` class, we can create `DEFAULT_SENSOR_RANGE` variable and assign this value. You can give another name if you wish.

Reply:
You are right, I was thinking of a commit or even a pull request on its own solely focusing on icing the code quality by delegating to constant variables, deleting extra spaces or adding the ones required and optimizing imports. I think it's my mistake not stating those plans under the pull request description. Anyway, this issue here is the perfect candidate for the hotfixes after the approval of this pull request.